plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkModifyJob.java

```diff
@@ -94,15 +96,20 @@ protected final IStatus run(IProgressMonitor monitor) {
 
     try {
       return modifySdk(monitor);
+    } catch (CloudSdkVersionFileException | CloudSdkNotFoundException ex) {
+      IStatus status = StatusUtil.error(CloudSdkModifyJob.class, ex.getMessage(), ex);
+      return status;
     } finally {
       cloudSdkLock.writeLock().unlock();
     }
   }
 
-  protected abstract IStatus modifySdk(IProgressMonitor monitor);
+  protected abstract IStatus modifySdk(IProgressMonitor monitor)
```

Comment:
To be consistent (with the fact that subclasses implementing this take care of all exceptions and return a final status), let's not change this. Catch these exception in `CloudSdkInstallJob`, which is what you did exactly in `CloudSdkUpdateJob`.